shopinvader/services/service.py

```diff
@@ -120,3 +121,23 @@ def _get_openapi_default_parameters(self):
             }
         )
         return defaults
+
+    @property
+    def shopinvader_session_data(self):
+        """
+        A a mutable dictionary which is injected in service responses.
+
+        Keys supported by the locomotive frontend include:
+        - store_cache
+        - set_session
+        """
+        if not hasattr(request, "_shopinvader_session_data"):
+            request._shopinvader_sessiondata = {}
+        return request._shopinvader_sessiondata
+
+    def dispatch(self, method_name, _id=None, params=None):
+        res = super().dispatch(self, method_name, _id=_id, params=params)
+        shopinvader_session_data = self.shopinvader_session_data
+        if shopinvader_session_data:
+            res.update(shopinvader_session_data)
```

Comment:
IMO the way to process session data (and its content) will depend of the kind of client consuming the services. Do we have to provide this implementation into the base addon? Into the current context, this code is specific to locomotive.

Reply:
@lmignon yes you are right. Yet at this stage all the logic about set_session and store_cache is in the base shopinvader module. So it's a bigger refactoring.

Reply:
IMO yes and no :innocent: If you think about this as a generic tool, then being able to provide values to be set on the session on the frontend service is a must have. No?
If we are talking about specific key names: I agree :)